cmd/startupapicheck/pkg/factory/factory.go

```diff
@@ -72,14 +72,18 @@ func New(ctx context.Context, cmd *cobra.Command) *Factory {
 	kubeConfigFlags.AddFlags(cmd.Flags())
 	cmd.RegisterFlagCompletionFunc("namespace", validArgsListNamespaces(ctx, f))
 
-	// Setup a PreRun to populate the Factory. Catch the existing PreRun command
+	// Setup a PreRunE to populate the Factory. Catch the existing PreRunE command
 	// if one was defined, and execute it second.
-	existingPreRun := cmd.PreRun
-	cmd.PreRun = func(cmd *cobra.Command, args []string) {
-		util.CheckErr(f.complete())
-		if existingPreRun != nil {
-			existingPreRun(cmd, args)
+	existingPreRunE := cmd.PreRunE
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		if err := f.complete(); err != nil {
+			return err
 		}
+
+		if existingPreRunE != nil {
+			return existingPreRunE(cmd, args)
+		}
+		return nil
```

Comment:
question: The PR description for this PR says that the bug it's fixing was introduced in https://github.com/cert-manager/cert-manager/pull/6562 - but that PR doesn't touch `cmd/startupapicheck`. Why is this file changing?

Reply:
I changed all the PreRun occurrences. If we are not mixing PreRunE and PreRun there is no problem.
I'll check specifically for startupapicheck and remove this change if not required.

Reply:
fixed.